src/github.com/getlantern/enproxy/conn_impl.go

```diff
@@ -38,12 +37,18 @@ func Dial(addr string, config *Config) (*Conn, error) {
 		return nil, fmt.Errorf("Unable to dial proxy: %s", err)
 	}
 
+	go c.processWrites()
+	go c.processReads()
 	go c.processRequests(proxyConn)
 
-	return c, nil
+	return idletiming.Conn(c, c.config.IdleTimeout, func() {
+		c.Close()
+		// Close the initial proxyConn just in case
+		proxyConn.conn.Close()
+	}), nil
```

Comment:
enproxy.conn used to do its own idle timing.  To simplify the code, I removed that and wrap it with an idletiming.Conn instead.  idletiming is well unit tested, so it's safer to use than the old enproxy logic which wasn't as thoroughly unit tested.  Also, it's nice to separate these concerns.